### PR TITLE
Package Version update to 2.9.2

### PR DESCRIPTION
**Pull request by psteyer**
Updating package due to being effected by multiple CVE's. 

CVE-2023-6753
CVE-2023-6831
CVE-2023-6018
CVE-2023-6974
CVE-2023-6975
CVE-2023-6976
CVE-2023-6940

Jira: https://anaconda.atlassian.net/browse/PKG-3694

### Changes
- updated version number
- updated SHA

### CODE REVIEW

**Review by JeanChristopheMorinPerso — CHANGES_REQUESTED** [review]
The list of dependencies changed between 2.6.0 and 2.9.2. The list for the "snikky" variant can be found in https://github.com/mlflow/mlflow/blob/v2.9.2/requirements/skinny-requirements.txt and the list for the non snikky in https://github.com/mlflow/mlflow/blob/v2.9.2/requirements/core-requirements.txt. You can also find the stuff that is in `run_constrained` in https://github.com/mlflow/mlflow/blob/v2.9.2/requirements/extra-ml-requirements.txt.

**Comment by psteyer** [comment]
I noticed there is also a difference between our `importlib-metadata` naming and the `skinny-requirments.txt`  ->`importlib_metadata`. Does this make a difference?

**Comment by JeanChristopheMorinPerso** [comment]
No. The package names in Python are all normalized, which means that `-` is the same as `_`. Note that this behavior is specific to the python ecosystem and that it's different in the conda world. With conda `-` is different than `_`.

**Comment by psteyer** [comment]
> The list of dependencies changed between 2.6.0 and 2.9.2. The list for the "snikky" variant can be found in https://github.com/mlflow/mlflow/blob/v2.9.2/requirements/skinny-requirements.txt and the list for the non snikky in https://github.com/mlflow/mlflow/blob/v2.9.2/requirements/core-requirements.txt. You can also find the stuff that is in `run_constrained` in https://github.com/mlflow/mlflow/blob/v2.9.2/requirements/extra-ml-requirements.txt.

I updated some requirement pinnings, but I couldn't find anything in https://github.com/mlflow/mlflow/blob/v2.9.2/requirements/extra-ml-requirements.txt

File: recipe/meta.yaml [hunk 1]
```diff
@@ -55,12 +55,12 @@ requirements:
...
     - packaging <24
-    - importlib-metadata <7,>=3.7.0,!=4.7.0
+    - importlib-metadata <8,>=3.7.0,!=4.7.0
     - sqlparse >=0.4.0,<1.0
 {% if mlflow_variant != "skinny" %}
     - alembic <2,!=1.10.0
     - docker-py >=4.0.0,<7.0
-    - flask <3.0
+    - flask <4.0
     - numpy <2.0
```
Comment (danpetry) [inline recipe/meta.yaml:64]: you could put `- numpy {{ numpy }}`, see cbc.yaml for the reason

File: recipe/meta.yaml [hunk 2]
```diff
@@ -69,7 +69,7 @@ requirements:
     - gunicorn <22  # [not win]
     - waitress <3  # [win]
     - scikit-learn <2.0
-    - pyarrow <13,>=4.0.0
+    - pyarrow <15,>=4.0.0
     - markdown >=3.3,<4.0
     - jinja2 <4,>=2.11  # [not win]
     - jinja2 <4,>=3.0  # [win]
```
Comment (danpetry) [inline recipe/meta.yaml:75]: pytorch_lightning >=1.8.1
https://github.com/mlflow/mlflow/blob/6ca72469b289e77acc2f1201ca39237fc025c090/requirements/extra-ml-requirements.txt#L16

Comment (danpetry) [reply recipe/meta.yaml:75]: also shap >=0.42.1

Comment (danpetry) [reply recipe/meta.yaml:75]: and torch should be pytorch, I think torch is the pip name

**Comment by danpetry** [comment]
packages in requirements.txt in repos refer to packages on pypi. Our recipes refer to conda packages. The names are usually the same but sometimes a bit different, so it's worth keeping an eye out for packages that have similar names when comparing our recipes with upstream repos

**Comment by danpetry** [comment]
and yeah, always worth checking the deps haven't changed if you upgrade the version number. You can do a diff upstream via the pull request button (but not actually opening a PR) and checking whichever file the requirements are expressed in

File: recipe/meta.yaml (hunk 1 above)
Comment (skupr-anaconda) [reply recipe/meta.yaml:64]: `- numpy {{ numpy }}` variant is only for `host`

File: recipe/meta.yaml (hunk 2 above)
Comment (JeanChristopheMorinPerso) [reply recipe/meta.yaml:75]: @danpetry `pytorch_lightning` should be removed. `lightning>=1.8.1` should be added and `openai<1.0` too.

**Comment by skupr-anaconda** [comment]
An error on linux and win but not on osx:
```
conda.CondaError: Downloaded bytes did not match Content-Length
  url: https://pypi.io/packages/source/m/mlflow/mlflow-2.9.2.tar.gz
  target_path: /src_cache/src_cache/mlflow-2.9.2_25f2ad3d0b.tar.gz
  Content-Length: 18655983
```

**Review by JeanChristopheMorinPerso — APPROVED** [review]

**Review by danpetry — APPROVED** [review]

**Pull request merged**